compiler/ml/error_message_utils.ml

```diff
@@ -484,18 +580,24 @@ let path_to_jsx_component_name p =
 
 let get_jsx_component_props
     ~(extract_concrete_typedecl : extract_concrete_typedecl) env ty p =
-  match Path.last p with
-  | "props" -> (
-    try
-      match extract_concrete_typedecl env ty with
-      | ( _p0,
-          _p,
-          {Types.type_kind = Type_record (fields, _repr); type_attributes} )
-        when attributes_include_jsx_component_props type_attributes ->
-        Some {props_record_path = p; fields}
-      | _ -> None
-    with _ -> None)
-  | _ -> None
+  match p with
+  | Path.Pdot (Path.Pident {Ident.name = jsx_module_name}, "fragmentProps", _)
+    when Some jsx_module_name = !configured_jsx_module ->
+    Some {props_record_path = p; fields = []; jsx_type = `Fragment}
```

Comment:
This logic isn't perfect, and at some point we probably need to figure out a better approach for matching on "what is a JSX component" in order to give better error messages. But for now with the state we're in, this is an improvement.